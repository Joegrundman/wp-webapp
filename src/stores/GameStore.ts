import { URL_MAP_EUR, URL_MAP_PAC } from 'Constants/ui-constants'
import { observable } from 'mobx'

class GameStore {

  @observable public theater: string = URL_MAP_EUR
  @observable public showDialogPanel: boolean = false
  @observable public currentActiveDialog: string = ''

  public setTheater (theater: string): void {
    this.theater = theater
  }

  public toggleTheater (): void {
    if (this.theater === URL_MAP_EUR) {
      this.theater = URL_MAP_PAC
    } else {
      this.theater = URL_MAP_EUR
    }
  }

  public toggleDialogPanel (): void {
    this.showDialogPanel = !this.showDialogPanel
  }

  public openDialogPanel (): void {
    this.showDialogPanel = true
  }

  public handleDialogButton(dialog: string): void {
    if(this.showDialogPanel && this.currentActiveDialog !== dialog) {
      this.setCurrentActiveDialog(dialog)
    } else if(this.showDialogPanel && this.currentActiveDialog === dialog) {
      this.closeDialogPanel()
      this.setCurrentActiveDialog('')
    } else {
      this.openDialogPanel()
      this.setCurrentActiveDialog(dialog)
    }  
  }

  public closeDialogPanel (): void {
    this.showDialogPanel = false
  }

  public setCurrentActiveDialog (dialog: string): void {
    this.currentActiveDialog = dialog
  }
}

export default GameStore