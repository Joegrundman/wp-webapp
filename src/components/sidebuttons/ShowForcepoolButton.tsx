import showForcepoolImg from 'Assets/Buttons/forcepool.png'
import Sidebutton from 'Atoms/sidebutton/Sidebutton'
import * as React from 'react'
import GameStore from 'Stores/GameStore'

interface ISidebarProps {
  store: GameStore
}

const ShowForcepoolBtn: React.SFC<ISidebarProps> = ({ store }): JSX.Element => {

  const handleAction = (): void => {
    store.handleDialogButton('forcepool')   
  }

  return (
  <div>
    <Sidebutton image={showForcepoolImg} action={handleAction}/>
  </div>
)}

export default ShowForcepoolBtn
