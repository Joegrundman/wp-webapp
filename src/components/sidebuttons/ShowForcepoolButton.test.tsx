import { shallow, ShallowWrapper } from 'enzyme'
import * as React from 'react'
import GameStore from 'Stores/GameStore'
import ShowForcepoolButton from './ShowForcepoolButton'

describe('<ShowForcepoolButton />', (): void => {
  const store: GameStore = new GameStore()
  it('should render as expected', (): void => {
    const component: ShallowWrapper = shallow(<ShowForcepoolButton store={store} />)
    expect(component).toMatchSnapshot()
  })

  it('should have an action that updates the store by setting the correct dialog', (): void => {
    store.handleDialogButton = jest.fn()

    const component: ShallowWrapper = shallow(<ShowForcepoolButton store={store} />)
    const action = component
      .find('Sidebutton').first().prop('action')
    // @ts-ignore
    action()

    expect(store.handleDialogButton).toHaveBeenCalledWith('forcepool')
  })
})
