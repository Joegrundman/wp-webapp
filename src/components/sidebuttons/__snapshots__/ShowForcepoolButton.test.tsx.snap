// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowForcepoolButton /> should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowForcepoolBtn
    store={
      GameStore {
        "currentActiveDialog": "",
        "showDialogPanel": false,
        "theater": "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
        Symbol(mobx did run lazy initializers): true,
        Symbol(mobx administration): ObservableObjectAdministration$$1 {
          "defaultEnhancer": [Function],
          "keysAtom": Atom$$1 {
            "diffValue": 0,
            "isBeingObserved": false,
            "isPendingUnobservation": false,
            "lastAccessedBy": 0,
            "lowestObserverState": 2,
            "name": "GameStore@1.keys",
            "observers": Set {},
          },
          "name": "GameStore@1",
          "target": [Circular],
          "values": Map {
            "theater" => "http://res.cloudinary.com/druzhkwmt/image/upload/v1456179368/b3oj0apypmsiu1zsypwl.jpg",
            "showDialogPanel" => false,
            "currentActiveDialog" => "",
          },
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Sidebutton
        action={[Function]}
        image={undefined}
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "action": [Function],
        "image": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Sidebutton
          action={[Function]}
          image={undefined}
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "action": [Function],
          "image": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
